src/responder/pam/pamsrv_gssapi.c

```diff
@@ -83,6 +85,102 @@ static bool pam_gssapi_should_check_upn(struct pam_ctx *pam_ctx,
     return pam_ctx->gssapi_check_upn;
 }
 
+static int pam_gssapi_check_indicators(struct pam_ctx *pam_ctx,
```

Comment:
Hi,

please add a comment which explains the special meaning of `EINVAL` with respect to access control. 

`domain` is only used to assign `map`, why not just pass `map` directly?

`pam_ctx` is a long living context which will be freed only if the process exits. `cli_ctx` from the caller might be more suitable. I think a generic `TALLOC_CTX *mem_ctx` would fit better here.

bye,
Sumit

Reply:
Hi,

after reading the function more carefully I wonder if `EINVAL` should really be treated as access allowed as suggested by the comment in the caller. Do you by chance mean `ENOENT` ?

bye,
Sumit

Reply:
An idea was to differentiate cases when internal processing fails and we really cannot decide whether it is a broken configuration. The comment in the caller should have really said ENOENT, indeed. What is a typical pattern in SSSD for the situations when you cannot decide whether a broken setup should allow or deny access? I'd probably tend to use denial here.